tests/python_client/testcases/test_query.py

```diff
@@ -2854,3 +2792,125 @@ def test_count_expression_comparative(self):
+        collection_w.load()
+        # 2. search iterator
+        expr = "int64 >= 0"
+        collection_w.query_iterator(expr, limit=limit, offset=offset,
+                                    check_task=CheckTasks.check_query_iterator,
+                                    check_items={"count": ct.default_nb - offset,
+                                                 "limit": limit})
+
+    @pytest.mark.tags(CaseLabel.L1)
+    @pytest.mark.parametrize("limit", [10, 100, 777, 2000])
+    def test_query_iterator_with_different_limit(self, limit):
+        """
+        target: test query iterator normal
+        method: 1. query iterator
+                2. check the result, expect pk
+        expected: query successfully
+        """
+        # 1. initialize with data
+        offset = 500
+        collection_w = self.init_collection_general(prefix, True, is_index=False)[0]
+        collection_w.create_index(ct.default_float_vec_field_name, {"metric_type": "L2"})
+        collection_w.load()
+        # 2. search iterator
+        expr = "int64 >= 0"
+        collection_w.query_iterator(expr, limit=limit, offset=offset,
+                                    check_task=CheckTasks.check_query_iterator,
+                                    check_items={"count": ct.default_nb - offset,
+                                                 "limit": limit})
+
+    @pytest.mark.tags(CaseLabel.L0)
+    @pytest.mark.parametrize("limit", [100, 500])
+    @pytest.mark.parametrize("auto_id", [True, False])
+    def test_query_iterator_empty_expr(self, auto_id, limit):
+        """
+        target: test query iterator with empty expression
+        method: query iterator empty expression with a limit
+        expected: return topK results by order
+        """
+        # 1. initialize with data
+        collection_w, _, _, insert_ids = self.init_collection_general(prefix, True, auto_id=auto_id)[0:4]
+
+        # 2. query with limit
+        collection_w.query_iterator("", limit=limit,
+                                    check_task=CheckTasks.check_query_iterator,
+                                    check_items={"limit": limit,
+                                                 "count": ct.default_nb,
+                                                 "exp_ids": insert_ids})
+
+    @pytest.mark.tags(CaseLabel.L2)
+    @pytest.mark.parametrize("offset", [100, 1000])
+    @pytest.mark.parametrize("limit", [500, 1000])
+    def test_query_iterator_expr_empty_with_random_pk_pagination(self, limit, offset):
+        """
+        target: test query iterator with empty expression
+        method: create a collection using random pk, query empty expression with a limit
+        expected: return topK results by order
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, with_json=False)[0]
+
+        # 2. generate unordered pk array and insert
```

Comment:
Function "init_collection_general " could insert random pk directly setting random_primary_key as True.

Reply:
I have modified the case using random_primary_key=True.